elements/lisk-p2p/test/unit/peer_pool.ts

```diff
@@ -160,6 +161,76 @@ describe('peerPool', () => {
 		});
 	});
 
+	describe('#request', () => {
+		let caughtError: Error;
+		beforeEach(async () => {
+			(peerPool['_peerSelectForRequest'] as any) = sandbox
+				.stub()
+				.returns([] as ReadonlyArray<P2PPeerInfo>);
+			try {
+				await peerPool.request({ procedure: 'proc', data: 123 });
+			} catch (err) {
+				caughtError = err;
+			}
+		});
+
+		it('should call _peerSelectForRequest with all the necessary options', async () => {
+			expect(peerPool['_peerSelectForRequest']).to.be.calledWith({
```

Comment:
Prefer to write `(peerPool as any)._peerSelectForRequest`. @mitsujutsu told me to follow this approach. Please, review every case from this pull request.